llvm/lib/IR/PrintPasses.cpp

```diff
@@ -33,6 +33,31 @@ static cl::opt<bool> PrintAfterAll("print-after-all",
                                    llvm::cl::desc("Print IR after each pass"),
                                    cl::init(false), cl::Hidden);
 
+static cl::list<std::string>
```

Comment:
There is a lot duplication between the print code and the dump code, though there are differences, it might be nice to try to combine them a little more.